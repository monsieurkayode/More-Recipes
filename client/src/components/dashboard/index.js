/* jshint esversion: 6 */
import UserProfile from './UserProfile';
import UserRecipe from './UserRecipe';
import WelcomeDisplay from './WelcomeDisplay';
import DashboardPanel from './DashboardPanel';
import SideNavDashboard from './SideNavDashboard';
import UserFavoriteRecipe from './UserFavoriteRecipe';
import UserNotification from './UserNotification';

export {
  UserProfile,
  UserRecipe,
  UserFavoriteRecipe,
  UserNotification,
  WelcomeDisplay,
  SideNavDashboard,
  DashboardPanel
};
