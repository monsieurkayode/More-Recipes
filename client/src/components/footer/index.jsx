import React from 'react';

/**
 * Footer
 * @function Footer
 *
 * @param {void} void
 *
 * @return {JSX} JSX
 */
const Footer = () => (
  <footer className="page-footer blue-grey darken-2">
    <div className="row">
      <div className="col l3 social-icon-container">
        <ul>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-github teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-facebook teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-twitter teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-google teal-text" />
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div className="footer-copyright">
      <div className="container">
        Â© Andela 2018
        <a
          className="grey-text text-lighten-4 right"
          href="/"
        >
        Created by Kayode Adeola
        </a>
      </div>
    </div>
  </footer>
);

export default Footer;
