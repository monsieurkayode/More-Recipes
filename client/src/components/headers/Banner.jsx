import React from 'react';
import bannerImage from '../../../assets/css/img/banner2.jpg';

/**
 * Banner
 * @function Banner
 *
 * @param {void} void
 *
 * @return {JSX} JSX
 */
const Banner = () => (
  <div id="banner" className="hide-on-small-only">
    <div className="card z-depth-0">
      <div className="card-image">
        <img className="responsive-img" src={bannerImage} alt="" />
      </div>
    </div>
  </div>
);

export default Banner;
