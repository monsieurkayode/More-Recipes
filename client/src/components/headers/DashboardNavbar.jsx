import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'proptypes';

import { UserMenu } from './';
import resetPage from '../../utils/resetPage';

/**
 * DashboardNavbar
 * @function DashboardNavbar
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const DashboardNavbar = (props) => {
  const brandClass = 'brand-logo font-effect-3d-float logo';
  return (
    <div className="navbar-fixed">
      <nav className="header">
        <div className="nav-wrapper">
          <Link
            to="/"
            onClick={resetPage}
            className={`${brandClass}`}
          >
              More-Recipes
          </Link>
          <a
            href=""
            data-activates="dashboard"
            className="button-collapse"
          >
            <i className="material-icons">menu</i>
          </a>
          <ul className="right hide-on-med-and-down return">
            <li>
              <a className="dropdown-button" data-activates="user-dropdown">
                <span
                  className="fa fa-lg fa-user-circle"
                /> {props.user.username} <span
                  className="fa fa-lg fa-caret-down"
                />
              </a>
              <UserMenu logoutAction={props.logoutAction} />
            </li>
          </ul>
        </div>
      </nav>
    </div>
  );
};

DashboardNavbar.defaultProps = {
  user: {}
};

DashboardNavbar.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string
  }),
  logoutAction: PropTypes.func.isRequired
};

export default DashboardNavbar;
