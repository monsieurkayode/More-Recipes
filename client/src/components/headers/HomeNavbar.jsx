import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'proptypes';

import { logoutAction } from '../../actions';
import { UserMenu, SearchBar } from './';
import resetPage from '../../utils/resetPage';

const HomeNavbar = (props) => {
  const brandClass = 'brand-logo font-effect-3d-float logo';
  return (
    <div className="navbar-fixed">
      <nav className="header">
        <div className="nav-wrapper">
          <Link
            to="/"
            onClick={resetPage}
            className={`${brandClass}`}
            style={{ whiteSpace: 'nowrap' }}
          >
            More-Recipes
          </Link>
          <a
            href=""
            data-activates="navlink"
            className="button-collapse"
          >
            <i className="material-icons">menu</i>
          </a>
          <ul className="right hide-on-med-and-down">
            <li>
              <SearchBar />
            </li>
            <li>
              <Link to="/" onClick={resetPage}>
                <i className="fa fa-home" /> Home
              </Link>
            </li>
            <li>
              <Link to="/recipes/new" id="new-post">
                <i className="fa fa-pencil-square-o" /> Add Recipe
              </Link>
            </li>
            <li className="right hide-on-med-and-down return">
              <a className="dropdown-button" data-activates="user-dropdown">
                <span
                  className="fa fa-lg fa-user-circle"
                /> {props.user.username} <span
                  className="fa fa-lg fa-caret-down"
                />
              </a>
              <UserMenu {...props} />
            </li>
          </ul>
        </div>
      </nav>
    </div>
  );
};

const mapStateToProps = ({ signinState }) => ({ user: signinState.user });

HomeNavbar.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string
  }).isRequired
};

export default connect(mapStateToProps, { logoutAction })(HomeNavbar);
