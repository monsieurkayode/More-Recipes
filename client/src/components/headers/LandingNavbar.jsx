import React from 'react';
import { Link } from 'react-router-dom';

import { SearchBar } from './';

/**
 * LandingNavBar
 * @function LandingNavBar
 *
 * @param {void} void
 *
 * @return {JSX} JSX
 */
const LandingNavBar = () => (
  <div className="navbar-fixed">
    <nav className="header">
      <div className="nav-wrapper">
        <Link
          to="/"
          className="brand-logo font-effect-3d-float logo hide-on-small-only"
        >
        More-Recipes
        </Link>
        <a
          href=""
          data-activates="navlink"
          className="button-collapse"
        >
          <i className="material-icons">menu</i>
        </a>
        <ul className="right hide-on-med-and-down">
          <li>
            <SearchBar />
          </li>
          <li>
            <Link to="/">
              <i className="fa fa-heart" /> Top Recipes
            </Link>
          </li>
          <li>
            <Link to="/recipes/new" id="new-post">
              <i className="fa fa-pencil-square-o" /> New Post
            </Link>
          </li>
          <li>
            <Link
              to="/signin"
              className="waves-effect waves-light btn navbtn"
            >
              Login
            </Link>
          </li>
          <li>
            <Link
              to="/signup"
              className="waves-effect waves-light btn navbtn"
            >
              Sign Up
            </Link>
          </li>
        </ul>
      </div>
    </nav>
  </div>
);

export default LandingNavBar;
