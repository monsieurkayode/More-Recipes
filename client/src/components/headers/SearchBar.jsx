import React, { Component } from 'react';

/**
 * @summary -SearchBar class declaration
 * @class SearchBar
 * @extends {Component}
 */
class SearchBar extends Component {
  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    return (
      <form id="nav-search">
        <input
          className="white input-fa"
          type="search"
          placeholder="&#xf002; Search"
        />
      </form>
    );
  }
}

export default SearchBar;
