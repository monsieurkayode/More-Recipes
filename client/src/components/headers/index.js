/* jshint esversion: 6 */
import Banner from './Banner';
import DashboardNavbar from './DashboardNavbar';
import HomeNavbar from './HomeNavbar';
import LandingNavbar from './LandingNavbar';
import UserMenu from './UserMenu';
import SearchBar from './SearchBar';

export {
  Banner,
  DashboardNavbar,
  HomeNavbar,
  LandingNavbar,
  UserMenu,
  SearchBar
};
