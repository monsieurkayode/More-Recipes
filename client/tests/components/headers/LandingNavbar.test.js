/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { LandingNavbar } from '../../../src/components/headers';


const setup = () => {
  const wrapper = mount(
    <BrowserRouter>
      <LandingNavbar />
    </BrowserRouter>
  );
  return {
    wrapper
  };
};

describe('The LandingNavbar Component', () => {
  it('should render correctly', (done) => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });
});
