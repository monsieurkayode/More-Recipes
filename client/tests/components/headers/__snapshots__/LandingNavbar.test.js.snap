// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LandingNavbar Component should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LandingNavBar>
      <div
        className="navbar-fixed"
      >
        <nav
          className="header"
        >
          <div
            className="nav-wrapper"
          >
            <Link
              className="brand-logo font-effect-3d-float logo hide-on-small-only"
              replace={false}
              to="/"
            >
              <a
                className="brand-logo font-effect-3d-float logo hide-on-small-only"
                href="/"
                onClick={[Function]}
              >
                More-Recipes
              </a>
            </Link>
            <a
              className="button-collapse"
              data-activates="navlink"
              href=""
            >
              <i
                className="material-icons"
              >
                menu
              </i>
            </a>
            <ul
              className="right hide-on-med-and-down"
            >
              <li>
                <SearchBar>
                  <form
                    id="nav-search"
                  >
                    <input
                      className="white input-fa"
                      placeholder="ï€‚ Search"
                      type="search"
                    />
                  </form>
                </SearchBar>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-heart"
                    />
                     Top Recipes
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  id="new-post"
                  replace={false}
                  to="/recipes/new"
                >
                  <a
                    href="/recipes/new"
                    id="new-post"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-pencil-square-o"
                    />
                     New Post
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  className="waves-effect waves-light btn navbtn"
                  replace={false}
                  to="/signin"
                >
                  <a
                    className="waves-effect waves-light btn navbtn"
                    href="/signin"
                    onClick={[Function]}
                  >
                    Login
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  className="waves-effect waves-light btn navbtn"
                  replace={false}
                  to="/signup"
                >
                  <a
                    className="waves-effect waves-light btn navbtn"
                    href="/signup"
                    onClick={[Function]}
                  >
                    Sign Up
                  </a>
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </LandingNavBar>
  </Router>
</BrowserRouter>
`;
